modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts

```diff
@@ -81,7 +81,11 @@ const applyDataToElement = (editor: Editor, tableElm: HTMLTableElement, data: Ta
     styles['border-style'] = data.borderstyle;
   }
 
-  attrs.style = dom.serializeStyle({ ...Options.getDefaultStyles(editor), ...styles });
+  const fullStyleAttribute = { ...Options.getDefaultStyles(editor), ...styles };
```

Comment:
This seems like the wrong approach to take... I've looked into it more and the cause is that `table_default_styles` has `width: 100%` on it when `table_style_by_css` is `false`. That seems like the root bug here instead, as we're not supposed to be using CSS we're supposed to be using attributes wherever possible.

As such, I'd suggest changing the defaults instead so that it puts the width in via attributes when set to `false` instead of via CSS>

Reply:
Since the default option doesn't also provide an attribute object, if we go with this, do we want to create some kind of default attribute object in turn, or strip attriutes from the css, since we still want some sort of default for them?

Reply:
Not sure what you mean sorry? There is `table_default_attributes`?

Reply:
I am blind, that is what I mean. Blind and missing what is staring me right in the face.